tiatoolbox/tools/patchextraction.py

```diff
@@ -241,10 +243,7 @@ def default_sel_func(reader: wsireader.VirtualWSIReader, coord: np.ndarray):
             """Accept coord as long as its box contains bits of mask."""
             roi = reader.read_bounds(
                 coord,
-                resolution=reader.info.mpp if resolution is None else resolution,
-                units="mpp" if units is None else units,
                 interpolation="nearest",
-                coord_space="resolution",
```

Comment:
To fix this, replace 
https://github.com/TissueImageAnalytics/tiatoolbox/blob/8581a9196d8b5281eee30fdf1c05268deddf211d/tiatoolbox/tools/patchextraction.py#L171-L184

and rely on using  https://github.com/TissueImageAnalytics/tiatoolbox/blob/333e20b578e41f2989be170b983e755f39f9ca2c/tiatoolbox/wsicore/wsireader.py#L489
to retrieve image HW at that specific target resolution.

Additionally, set https://github.com/TissueImageAnalytics/tiatoolbox/blob/8581a9196d8b5281eee30fdf1c05268deddf211d/tiatoolbox/tools/patchextraction.py#L155
to read at `coord_space="resolution"` also.

Reply:
Thanks for the suggestion @vqdang.
I just like to know why do you think "it's not good to return as coordinates with respect to baseline. I think we should use coordinates in the target resolution"?
Is it because of some performance (speed) or accuracy issues?

Reply:
Well, this is because of the conventional mental model when saying coordinates. For example, for me, when referring to 512x512 at 0.5 mpp, it means exactly that the expected coordinates are 512x512 at mpp=0.5, not 1024x1024 within baseline mpp=0.25 or any other Height x Width values that vary wrt to the baseline resolution. I expect this to be the same for many other people as well.

Reply:
Usually models are designed at a particular resolution which @vqdang is referring to as target resolution. And it is easier to run models if we know the coordinates at target resolution otherwise we may need conversion. So I would agree with @vqdang that it is better to use target resolution rather than baseline. @mostafajahanifar what do you think?